src/utilities/time.ts

```diff
@@ -47,23 +206,167 @@ function parseDurationString(durationString: string): ParsedDurationString | nev
     };
   }
 
-  const fullMicrosecondsRegex = new RegExp(`^${validDurationValueRegex}$`);
+  const isEpoch = validateTime(durationString, TimeTypes.EPOCH);
+  matches = isEpoch ? EPOCH_TIME.exec(durationString) : RELATIVE_TIME.exec(durationString);
+  if (matches !== null) {
+    const [, sign = '', days = '0', hours = '0', minutes = '0', seconds = '0', milliseconds = '0'] =
+      matches as RegExpExecArray;
+
+    const hoursNum = Number(hours);
```

Comment:
can you use `parseInt` here as well?

Reply:
we also have an existing DOY parsing function `parseDoyOrYmdTime` if it makes sense to use that. It seems like it does the exact same thing that you're doing here.

Reply:
Moved this to the `parseDOYDurationTime`